catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/transformer/CswActionTransformerProvider.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.ogc.csw.catalog.endpoint.transformer;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.Nullable;
+import javax.xml.namespace.QName;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manages a reference list of {@link CswActionTransformer}'s by mapping them to the QName's they
+ * apply to.
+ */
+public class CswActionTransformerProvider {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CswActionTransformerProvider.class);
+
+  private Map<QName, CswActionTransformer> transformerMap = new ConcurrentHashMap<>();
+
+  private final Map<String, QName> typeNameQNameMap = new HashMap<>();
+
+  public synchronized void bind(CswActionTransformer cswActionTransformer) {
+    if (cswActionTransformer == null) {
+      return;
+    }
+
+    List<QName> namespaces = cswActionTransformer.getQNames();
+    if (CollectionUtils.isEmpty(namespaces)) {
+      LOGGER.warn(
+          "Unable to bind a CswActionTransformer because it does not have one or more Qnames");
+      return;
+    }
+    for (QName namespace : namespaces) {
+      transformerMap.put(namespace, cswActionTransformer);
```

Comment:
❓If another CSW action transformer is registered with the same namespace it will overwrite the existing transformer. Is this the desired behavior?

Reply:
Hmm... yeah I think it's more intuitive to just not let the latest transformer bind at all and keep the existing one. This made me realize that I forgot to add unit tests for this class so I'll a test for this scenario and the other methods 👍